ts/component/dropdown.ts

```diff
@@ -44,16 +47,39 @@ export class Dropdown {
       dropdownMenuContainer.classList.remove("ring-2");
       dropdownClickout.classList.toggle("hidden");
     };
-    const invalidInput = () => {
-      dropdownMenuContainer.classList.add(
-        "ring-green-200",
-        "ring-2",
-        "ring-offset-1"
-      );
+    const toggleDropdownButton = () => {
+      // toggle whether or not a labeled button is removed
+      if (displayTemporary) {
+        dropdownButton.classList.toggle("hidden");
+      }
+      toggleMenu();
     };
     dropdownButton.onclick = toggleMenu;
     dropdownButton.innerHTML = makeDisplayText(displayLabel, defaultOption);
-    dropdownClickout.onclick = defaultOption ? toggleMenu : invalidInput;
+    // close dropdown menu by clicking outside of the menu or by hitting Esc
+    dropdownClickout.onclick = toggleDropdownButton;
+    document.addEventListener("keydown", (event) => {
+      // only remove the button in question if it is not already hidden
+      if (event.key === "Esc" || event.key === "Escape") {
+        const menuHidden = dropdownMenuContainer.classList.contains("hidden");
+        const clickoutHidden = dropdownClickout.classList.contains("hidden");
+        if (displayTemporary) {
+          // dropdown button is temporary, hide its corresponding elements if they are visible
+          if (
+            !dropdownButton.classList.contains("hidden") &&
+            !menuHidden &&
+            !clickoutHidden
+          ) {
+            toggleDropdownButton();
+          }
```

Comment:
Elements should be removed from DOM entirely instead of hiding them. Can just remove `dropdownContainer` to get rid of the entire thing